### PR TITLE
BAEL-2142 article code

### PR DESCRIPTION
**Pull request by codewarrior2000**
Two Java classes and one POM file update for BAEL-2142

### CODE REVIEW

File: java-strings/src/main/java/com/baeldung/string/WordIndexer.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            if (index != -1) {
+                indexes.add(index);
+            }else{
+                done = true;
+            }
+        }
+        return indexes;
+    }
+
+    public ArrayList findWordUpgrade(String textString, String word) {
```
Comment (KevinGilmore) [inline java-strings/src/main/java/com/baeldung/string/WordIndexer.java]: The two methods are identical. Isn't the first one supposed to be the naive approach?

File: java-strings/src/main/java/com/baeldung/string/WordIndexer.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        int index = -1;
+        ArrayList<Integer> indexes = new ArrayList<Integer>();
+        StringBuilder output = new StringBuilder();
+        boolean done = false;
+        String lowerCaseTextString = textString.toLowerCase();
+        String lowerCaseWord = word.toLowerCase();
+        int wordLength = word.length();
+
+        while(!done){
+            index = lowerCaseTextString.indexOf(lowerCaseWord, index + wordLength);  // Slight improvement
```
Comment (KevinGilmore) [inline java-strings/src/main/java/com/baeldung/string/WordIndexer.java]: What happens if the string ends with the word you're searching for? Would the next iteration of the loop cause an error if index + wordLength is greater than the last position of the string? Be sure to test this scenario.

Comment (codewarrior2000) [reply java-strings/src/main/java/com/baeldung/string/WordIndexer.java]: Thank you, Kevin. They are all very good points. 
Java's indexOf() method gracefully responds to "... if the string ends with the word you're searching for" (i.e., no error should happen).  I added a JUnit test anyway to test that scenario.

**Comment by codewarrior2000** [comment]
@KevinGilmore The merge conflict with java-strings/pom.xml is due to my addition of the following two dependencies for JUnit 5 however it could be that I don't have write access to this repository that I therefore cannot resolve the merge.

<dependency>
            <groupId>org.junit.jupiter</groupId>
            <artifactId>junit-jupiter-api</artifactId>
             <version>5.3.1</version>
             <scope>test</scope>
         </dependency>
 
         <dependency>
             <groupId>org.hamcrest</groupId>
             <artifactId>hamcrest-library</artifactId>
             <version>1.3</version>
            <scope>test</scope>
        </dependency>

File: java-strings/src/main/java/com/baeldung/string/WordIndexerUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package com.baeldung.string.searching;
```
Comment (KevinGilmore) [inline java-strings/src/main/java/com/baeldung/string/WordIndexerUnitTest.java]: Unit tests need to be in src/test/java, not in src/main/java

**Pull request closed without merging**